File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/user/domain/store/MenuCreator.java
package com.depromeet.threedollar.domain.user.domain.store;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class MenuCreator {

    public static Menu create(Store store, String name, String price, MenuCategoryType category) {
        return Menu.of(store, name, price, category);
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/store/MenuCreator.kt
package com.depromeet.threedollar.domain.user.domain.store

import com.depromeet.threedollar.domain.user.domain.ObjectMother

@ObjectMother
object MenuCreator {

    @JvmStatic
    fun create(
        store: Store,
        name: String,
        price: String,
        category: MenuCategoryType
    ): Menu {
        return Menu.builder()
            .store(store)
            .name(name)
            .price(price)
            .category(category)
            .build()
    }

}
